src/app/navigation-channel/navigation-event.ts

```diff
@@ -0,0 +1,16 @@
+export enum NavigationEventType {
+  selectTreeElement,    // some tree element is selected
+  closeEditor,          // an editor is closed
+  focusEditor           // an editor receives focus
+}
+
+export class NavigationEventContent {
+  path: string;
+  name: string;
+}
+
+export class NavigationEvent {
+  source: string;
+  type: NavigationEventType;
```

Comment:
I don't think that an enum is a good choice here. Better would be a qualified type. If we extract the code into their own repositories we don't want to add each event here.